### DISCUSSION TITLE
Clipping of large models

### DISCUSSION THREAD

**Discussion by zivschneider** [google/model-viewer#3751] [Q&A]
Hello, I am working with large scale models the size of buildings and need to show them to scale. 
Having an issue with the camera clipping and cannot load the full model. Is there a way to adjust the clipping plane? 

Documentation of the issue 
[Here ](https://drive.google.com/file/d/1lnT_lh8mfe1PbULnm_azUoDk2Ky3WkOv/view?usp=sharing)
and [Here ](https://drive.google.com/file/d/16llMu4Kzq8gi9W5ge2t0QBKAsyoCNmZL/view?usp=sharing)

**Comment by elalish** [comment]
Is your trouble in 3D or AR mode? If AR, which mode/modes are you using or having trouble with?

**Reply by zivschneider** [reply]
only in AR mode, Object mode is fine. What do you mean by modes in AR? @elalish Thank you!

**Reply by zivschneider** [reply]
apologies, using "webxr scene-viewer quick-look" @elalish

**Reply by elalish** [reply]
We have no control over SceneViewer and QuickLook (they are apps outside the browser). However, in WebXR mode, we are in control. It appears from your videos that you're using Scene Viewer with occlusion enabled (it's ARCore estimating the scene depth, rather than a simple clipping plane). You can disable it in the upper-right and try that. You can also try WebXR mode (since it's first, it must have failed, most likely because you aren't serving on HTTPS?). We don't attempt occlusion in WebXR currently.

**Reply by zivschneider** [reply]
I am trying to use only webxr mode with HTTPS and still no luck. How do you disable AR Core? 
Thank you so much for your help @elalish !

**Reply by elalish** [reply]
Well, WebXR depends on ARCore, so I don't think you want to disable it entirely, just occlusion. Check your JS console for info on why WebXR isn't launching. As for SceneViewer, try all the little UI widgets, including the three-dot menu. I think you'll find the occlusion switch.

**Reply by zivschneider** [reply]
for webxr not loading I am getting a "Unble to render content: No mixed-reality browser detected" (on Chrome, iOS) 
I tried adding `?disable_occlusion=true ` as mentioned here  https://github.com/google/model-viewer/pull/1784 but no luck for disabling occlusion

**Reply by elalish** [reply]
Oh sorry, you didn't mention you were on iOS. Apple doesn't support WebXR, so QuickLook is all you have. I don't believe there's any customization of their experience possible (@grorg at Apple).

**Reply by zivschneider** [reply]
I see. I am testing on iOS but actually targeting Android for this project . 
I tested this code on Android but occlusion was still happening.. any suggestions for how to do this correctly for Scene Viewer? 
Thanks!

**Reply by elalish** [reply]
Okay, so go back and try my earlier suggestions on an Android phone and share your results here.

**Reply by zivschneider** [reply]
ok so we tested on Android. 
The console log: "No Mixed Reality browser detected" and then "Attempting to present in AR with webXR"

We couldn't find any UI settings to disable occlusion but I did add the param ?disable_occlusion=true 
Occlusion is disabled BUT we are also seeing a small scale of the model, not the 1:1 Fixed scale as before.

**Reply by elalish** [reply]
As long as "Attempting to present in AR with webXR" was the last log, then WebXR seems to be working. In that case Scene Viewer query params won't apply. The "small scale" is actually just the object being far away. Once you move your phone enough to find the ground, it should move up. You can instruct your users of this with our `ar-status` attribute: look at the `ar-prompt` [here](https://modelviewer.dev/examples/augmentedreality/index.html#webXR), for instance.
